src/Avalonia.Base/Media/Fonts/FontCollectionBase.cs

```diff
@@ -271,108 +315,69 @@ internal static Typeface GetImplicitTypeface(Typeface typeface, out string norma
+                var match = false;
+                if (TryParseEnumItem(s_fontStyles, token, out var newStyle))
                 {
-                    continue;
+                    style = newStyle;
+                    match = true;
                 }
-
-                familyName = familyName.Replace(" " + weightString, "").TrimEnd();
-
-                return true;
-            }
-
-            return false;
-        }
-
-        internal static bool TryGetStyle(ref string familyName, out FontStyle style)
-        {
-            style = FontStyle.Normal;
-
-            var tokenizer = new StringTokenizer(familyName, ' ');
-
-            tokenizer.ReadSpan();
-
-            while (tokenizer.TryReadString(out var styleString))
-            {
-                //Do not try to parse an integer
-                if (new StringTokenizer(styleString).TryReadInt32(out _))
+                else if (TryParseEnumItem(s_fontWeights, token, out var newWeight))
                 {
-                    continue;
+                    weight = newWeight;
+                    match = true;
                 }
-
-                if (!Enum.TryParse(styleString, true, out style))
+                else if (TryParseEnumItem(s_fontStretches, token, out var newStretch))
                 {
-                    continue;
+                    stretch = newStretch;
+                    match = true;
                 }
 
-                familyName = familyName.Replace(" " + styleString, "").TrimEnd();
-
-                return true;
+                if (match)
+                {
+                    // Carve out matched word from the normalized name.
+                    normalizedFamilyNameBuilder ??= new StringBuilder(normalizedFamilyName);
+                    normalizedFamilyNameBuilder.Remove(tokenizer.CurrentTokenIndex - totalCharsRemoved, token.Length);
+                    totalCharsRemoved += token.Length;
+                }
             }
 
-            return false;
+            // Get rid of any trailing spaces.
+            normalizedFamilyName = (normalizedFamilyNameBuilder?.ToString() ?? normalizedFamilyName).TrimEnd();
+
+            //Preserve old font source
+            return new Typeface(typeface.FontFamily, style, weight, stretch);
         }
 
-        internal static bool TryGetStretch(ref string familyName, out FontStretch stretch)
+        private static bool TryParseEnumItem<T>((string, T)[] mapping, ReadOnlySpan<char> token, out T value)
+            where T : struct, Enum
         {
-            stretch = FontStretch.Normal;
-
-            var tokenizer = new StringTokenizer(familyName, ' ');
-
-            tokenizer.ReadSpan();
-
-            while (tokenizer.TryReadString(out var stretchString))
+            foreach (var (key, candidate) in mapping)
             {
-                if (new StringTokenizer(stretchString).TryReadInt32(out _))
+                if (token.Equals(key.AsSpan(), StringComparison.OrdinalIgnoreCase))
```

Comment:
Why not use `Enum.TryParse<T>(ReadOnlySpan<char>, ...)` here? It would avoid maintaining the mappings manually.

A shim could be added to the existing `EnumHelper` class here for .NET Standard 2.0, falling back to the `string` overload in this case.

Reply:
Ah I wasn't aware of the `EnumHelper`. I agree, this would be better from a maintainability perspective.

If we are fine with the netstandard2.0 build target to do more allocations than the newer .NET targets, then I will add the shims.

Reply:
Yes, that's usually fine for .NET Standard. All `SpanHelpers` methods used by the `SpanStringTokenizer` are already falling back to the `string` overloads on that platform, effectively allocating.

If people want the best performance, they should target .NET 8.0+.